### ISSUE TITLE
How to make VCPKG copies dependencies pdb files in output directory

### ISSUE THREAD

**Issue by Belox86** [microsoft/vcpkg#37943]
### Discussed in https://github.com/microsoft/vcpkg/discussions/37828

<div type='discussions-op-text'>

<sup>Originally posted by **Belox86** March 29, 2024</sup>
Hi everyone,
i have multiple "internal" library which i'm trying to manage with vcpkg+cmake. 
Generally speaking everything is working properly. 
I have only one minor issue which i can't solve. When working in a project that uses my internal dependecies (in Windows with MSVC compiler) i can't have the pdbs of my internal dependencies copied along with the dlls.

The dependencies have the vcpkg_copy_pdbs() in the portfile and they properly export the pdbs in the local cache ..\build\vcpkg_installed\x64-windows\debug\bin along with the dlls. it just miss the final step to have them copied where the executable files is placed so that dependencies symbol can be load. 
Note that the dependecies dlls are copied from the local cache to the output directories and the program starts properly. Also the pdb of the project itself are in the output directory. Only the pdb of the dependecies are missing. 
What is the proper way to handle this? It seems strange to me that vcpkg handle everything and then it just miss the final copy. 
Thank you very much. 
Michele</div>

**Comment by FrankXie05** [comment]
@Belox86 Thanks for posting this issue. Could you please provide an example? For example: After installing the port, the dependency dll is missing. :)

**Comment by Belox86** [comment]
ok i will try to do an example:
My project has tinyxml2 as dependency which is available from the official vcpkg repository.
so in my vcpkg.json file i have:
`"dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name":"tinyxml2"
    }
  ]`
when i run `cmake `and `cmake --build` with `-config Debug` the port is successfully installed and my application compiled. 
In the folder `myProject\build\Debug` i find the `tinyxml2.dll` but not the `tinyxml2.pdb`.
`tinyxml2.pdb` is present only in the local vcpkg cache in  `myProject\build\vcpkg_installed\x64-windows\debug\bin` but not copied in the output directory along with its dll.
How can i have dependencies pdbs copy with dlls in output directory?

**Comment by jjYBdx4IL** [comment]
Same issue here. Cannot debug dynamically linked boost libs because pdb files are not where the debugger expects them to be (alongside the dlls copied into the output binary folder).

https://stackoverflow.com/questions/73101359/what-is-the-canonical-way-of-copying-vcpkg-library-pdb-files-into-your-build-fo

https://stackoverflow.com/questions/78242702/how-to-make-vcpkg-copies-dependencies-pdb-files-in-output-directory

**Comment by jjYBdx4IL** [comment]
I just had the same issue after relocating the vcpkg installation. Try clearing the package cache and rebuilding the package. There is no need for the dependency pdbs to get copied into the output folder. MSVC will find them at their isntalled location.

**Comment by FrankXie05** [comment]
cc @BillyONeal

**Comment by aline117** [comment]
I personally think that as long as the PDB file is generated at compile time, you need to copy the output directory

**Comment by devinfluencer** [comment]
Right, in my case I have the build running on a DevOps pipeline and need the pdbs copied to output as I create a nuget from the result.
Is there at least a workaround?

**Comment by derekpiasecki** [comment]
> Right, in my case I have the build running on a DevOps pipeline and need the pdbs copied to output as I create a nuget from the result. Is there at least a workaround?

The DLLs get copied by https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/msbuild/applocal.ps1, which is called by https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/msbuild/vcpkg.targets, and it doesn't look like there is a straightforward way to add in a PDB file copy to your output folder (other than MS changing Vcpkg).

However, there are ways to determine the [installed folder](https://learn.microsoft.com/en-us/vcpkg/reference/installation-tree-layout) from which they need to be copied, so you can do it yourself, ~such as mentioned in https://github.com/microsoft/vcpkg/issues/8106.~  It seems that workaround to find the path is not needed, at least not in manifest mode, as the PDBs are in the `$(VcpkgBin)` folder.
